### PR TITLE
PostgreSQL support

### LINKED ISSUE #938: PostgreSQL targets
thanks

### PR DESCRIPTION
**Pull request by Eugeny**
Fixes #938

### CODE REVIEW

File: warpgate-common/src/config/target.rs [hunk 1]
```diff
@@ -107,6 +107,24 @@ pub struct TargetMySqlOptions {
     pub tls: Tls,
 }
 
+#[derive(Debug, Deserialize, Serialize, Clone, Object)]
+pub struct TargetPostgresOptions {
+    #[serde(default = "_default_empty_string")]
+    pub host: String,
+
+    #[serde(default = "_default_mysql_port")]
```
Comment (NefixEstrada) [inline warpgate-common/src/config/target.rs:115]: This seems to be incorrect!

File: warpgate-protocol-postgres/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+#[async_trait]
+impl ProtocolServer for PostgresProtocolServer {
+    async fn run(self, address: SocketAddr) -> Result<()> {
+        let certificate_and_key = {
+            let config = self.services.config.lock().await;
+            let certificate_path = config
+                .paths_relative_to
+                .join(&config.store.mysql.certificate);
```
Comment (NefixEstrada) [inline warpgate-protocol-postgres/src/lib.rs:47]: This seems wrong

File: warpgate-protocol-postgres/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            let key_path = config.paths_relative_to.join(&config.store.mysql.key);
```
Comment (NefixEstrada) [inline warpgate-protocol-postgres/src/lib.rs:48]: This seems wrong

File: warpgate-protocol-postgres/src/lib.rs (hunk 2 above)
Comment (Eugeny) [reply warpgate-protocol-postgres/src/lib.rs:47]: fixed, thanks!

File: warpgate-protocol-postgres/src/lib.rs (hunk 3 above)
Comment (Eugeny) [reply warpgate-protocol-postgres/src/lib.rs:48]: fixed

**Pull request merged**